toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -224,16 +241,18 @@ func buildResultsSetToNodesSet(statesSet map[string]*BuildResult) (result map[st
 	return
 }
 
-func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (failedSRPMs map[string]*BuildResult, prebuiltSRPMs, prebuiltDeltaSRPMs, builtSRPMs map[string]bool, blockedSRPMs map[string]*pkggraph.PkgNode) {
-	failedSRPMs = make(map[string]*BuildResult)
-	prebuiltSRPMs = make(map[string]bool)
-	prebuiltDeltaSRPMs = make(map[string]bool)
-	builtSRPMs = make(map[string]bool)
-	blockedSRPMs = make(map[string]*pkggraph.PkgNode)
+func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (srpmData srpmBuildDataContainer) {
```

Comment:
How about returning a pointer ?
```suggestion
func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState) (srpmData *srpmBuildDataContainer) {
```

or maybe drop the explit named return all together :) [just an opinion]

```go
func getSRPMsState(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildState)  *srpmBuildDataContainer {
```

Reply:
Curious why you prefer the pointer, afaik a struct with non-primative types (lists/maps etc.) will be a lightweight copy that references the same underlying lists/maps without duplication.